pkg/sql/sem/builtins/geo_builtins.go

```diff
@@ -2285,6 +2285,51 @@ Note if geometries are the same, it will return the LineString with the minimum
 		},
 	),
 
+	//
+	// Geography operations
+	//
+	"st_geohash": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"geography", types.Geography},
+			},
+			ReturnType: tree.FixedReturnType(types.String),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				g := args[0].(*tree.DGeography)
+				ret, err := geogfn.GeoHash(g.Geography, geogfn.GeoHashNoPrecision)
+				if err != nil {
+					return nil, err
+				}
+				return tree.NewDString(ret), nil
+			},
+			Info: infoBuilder{
+				info: "Accepts a geographic expression and returns a GeoHash representation of the geometry with full precision if a point is provided, or with variable precision based on the size of the feature",
+			}.String(),
+			Volatility: tree.VolatilityImmutable,
+		},
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"geography", types.Geography},
+				{"precision", types.Int4},
```

Comment:
nit: use Int instead of IntFamily (it doesn't really matter in the context of CRDB, but for consistency we should use Int4)